src/test/PulsarMock.js

```diff
@@ -0,0 +1,242 @@
+export class PulsarClientMock {
+  static inMemoryStore = new Map();
```

Comment:
it would be really nice if we were using typscript so we could understand what goes into the Map,etc 
Please add comments explaining keys and values for every static field